lambdas/initialise-ipv-session/src/main/java/uk/gov/di/ipv/core/initialiseipvsession/InitialiseIpvSessionHandler.java

```diff
@@ -227,6 +251,76 @@ public APIGatewayProxyResponseEvent handleRequest(
         }
     }
 
+    @Tracing
+    private void validateAndStoreHMRCInheritedIdentity(
+            String userId, JWTClaimsSet claimsSet, IpvSessionItem ipvSessionItem)
+            throws RecoverableJarValidationException, ParseException {
+        JarClaims claims;
+        try {
+            claims =
+                    OBJECT_MAPPER.convertValue(
+                            claimsSet.getJSONObjectClaim(CLAIMS_CLAIM), JarClaims.class);
+        } catch (IllegalArgumentException | ParseException e) {
+            throw buildRecoverableException(
+                    OAuth2Error.INVALID_REQUEST_OBJECT.setDescription("Unable to process claims"),
+                    claimsSet);
+        }
+        if (claims == null) {
+            return;
+        }
+
+        var userInfo = claims.userInfo();
+        if (userInfo == null) {
+            return;
+        }
+
+        var inheritedIdentityJwtClaim = userInfo.inheritedIdentityClaim();
+        if (inheritedIdentityJwtClaim == null) {
+            return;
+        }
```

Comment:
These null checks are pretty ugly :( - shame Java doesn't let you do conditional access.

One slightly sneaky way of simplifying is leveraging optionals, for example:
```
var inheritedIdClaim = Optional.ofNullable(claims)
        .map(JarClaims::userInfo)
        .map(JarUserInfo::inheritedIdentityClaim)
        .map(InheritedIdentityJwtClaim::value)
        .flatMap((values) -> values.stream().findFirst());
```